### DISCUSSION TITLE
Help Wanted: Correct syntax for `podman run --entrypoint`

### DISCUSSION THREAD

**Discussion by MagicTrevor** [containers/podman#12513] [Q&A] [answered]
I am attempting to use `--entrypoint=` with `podman run` to run a binary file and pass it a path. From what I can find I should be doing it like this: `--entrypoint=["path/to/my/binary", "/path/to/file"]`, however this returns `Error: executable file '[/path/to/my/binary,/path/to/file]' not found in $PATH: no such file or directory: OCI runtime attempted to invoke a command that was not found`.

Is there something special in regards to being a rootless container compared to the Docker equivalent?

**Comment by mheon** [comment, marked as answer]
Have you tried quoting the entrypoint - `--entrypoint='["/part/one", "/part/two"]'`

The entrypoint itself is interpreted as a JSON array first, and if that fails to decode, we treat it as a literal path on the filesystem; it seems here that the JSON decoding is failing, though that seems like valid JSON.

**Reply by MagicTrevor** [reply]
Quoting the array seems to work. Thanks!

**Reply by rhatdan** [reply]
You were seeing a BASH failure.  Bash was interpreting the ["path/to/my/binary", "/path/to/file"].

**Reply by MagicTrevor** [reply]
Oh okay, the syntax I was using was based on what I found online so I assumed it would work.
